BE/src/main/java/com/codesquad/issuetracker/auth/service/GitHubLoginService.java

```diff
@@ -0,0 +1,101 @@
+package com.codesquad.issuetracker.auth.service;
+
+import com.codesquad.issuetracker.auth.response.AccessTokenResponse;
+import com.codesquad.issuetracker.auth.JwtProvider;
+import com.codesquad.issuetracker.domain.user.User;
+import com.codesquad.issuetracker.auth.domain.GitHubUser;
+import com.codesquad.issuetracker.domain.user.UserRepository;
+import com.codesquad.issuetracker.auth.request.AccessTokenRequest;
+import com.codesquad.issuetracker.auth.response.GitHubUserResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.env.Environment;
+import org.springframework.http.RequestEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class GitHubLoginService {
+
+    private final UserRepository userRepository;
+
+    private final JwtProvider jwtProvider;
+
+    private static final String GITHUB_ACCESS_TOKEN_URL = "https://github.com/login/oauth/access_token";
+    private static final String GITHUB_USER_URL = "https://api.github.com/user";
+
+    private final String GITHUB_CLIENT_ID;
+    private final String GITHUB_CLIENT_SECRETS;
+
+    private final String GITHUB_CLIENT_ID_IOS;
+    private final String GITHUB_CLIENT_SECRETS_IOS;
```

Comment:
```suggestion
    private static final String GITHUB_ACCESS_TOKEN_URL = "https://github.com/login/oauth/access_token";
    private static final String GITHUB_USER_URL = "https://api.github.com/user";
    
    private final UserRepository userRepository;
    private final JwtProvider jwtProvider;
    private final String githubClientId;
    private final String githubClientSecret;
    private final String iOSGithubClientId;
    private final String iOSGithubClientSecret;
```

Screaming snake case는 상수에만 사용해주세요.